dspace-server-webapp/src/main/java/org/dspace/app/rest/exception/DSpaceApiExceptionControllerAdvice.java

```diff
@@ -190,6 +191,12 @@ protected void MissingParameterException(HttpServletRequest request, HttpServlet
                           HttpStatus.BAD_REQUEST.value());
     }
 
+    @ExceptionHandler({ PasswordNotValidException.class})
+    protected void handlePasswordNotValidException(HttpServletRequest request, HttpServletResponse resp, Exception ex)
+        throws IOException {
+        sendErrorResponse(request, resp, null, ex.getMessage(), HttpStatus.UNPROCESSABLE_ENTITY.value());
```

Comment:
It's usually not good practice to send `ex.getMessage()` back via the REST API (you'll notice most of the methods in this class don't do that). While there can be exceptions... in this case, it looks like we could hardcode this message on this line to just say `"The new password to set is not valid"`, as that appears to be the only message ever set in the code.  This would let us keep the error text in one place (which makes it easier to translate/change later on), and simply call `throw new PasswordNotValidException()` where needed, without a message.